Localization improvements

afernandes opened the pull request:
Automatically register new words in the Localization_Resourse when navigating in English.

thiennn commented:
Hi @afernandes  thank you for the PR. Could you please summary your changes and why we need StaticData-DefaultLocalization

afernandes commented:
when browsing the site using the en-US culture, will be saved to new keywords in the Localization_Resourse table, so that it is possible to translate using the administrative area in an easy way.

The auto-registering method is **AutoRegisterNewString** in src/Modules/SimplCommerce.Module.Localization/EFStringLocalizer.cs

----
Also added was the log console of the new untranslated keywords in AngularJS in src/Modules/SimplCommerce.Module.Core/wwwroot/admin/common/translation-service.js

---
`<html lang="@(Context.Features.Get<IRequestCultureFeature>().RequestCulture.UICulture.Name.ToLowerInvariant())">`
in Layouts.cshtml for Chrome to correctly identify the site language.

---
```
@if(CultureInfo.CurrentCulture.Name != "en-US")
{
<script simpl-append-version="true" src="~/lib/bootstrap-star-rating/js/locales/@(CultureInfo.CurrentCulture.Name).js"></script>   
}
```
bootstrap-star-rating for all languages other than en-US

---
angular-locale_pt-br.js Updated

---
language selector for the cozastore theme.
Note: missing the images of the flags that I will send tomorrow.

---
added @Localizer and {{:: vm.translate.get ('X')}} in a few words that were missing.

---
added translateService in localization-form.js

---
changed STANDARD_CULTURE_ID to "en-US"
and changed vm.selectedCultureId = 'en-US'
Whereas all key words are written in en-us in the source code.

afernandes commented:
StaticData-DefaultLocalization contains most of the keywords used in the site, to facilitate translation into other languages using the administrative area.
StaticData-DefaultLocalization was created using AutoRegisterNewString.
I used as a base to create the language pt-BR

hishamco commented on src/Modules/SimplCommerce.Module.Core/Areas/Core/Views/Account/Login.cshtml:
```diff
@@ -67,7 +67,7 @@
                             for details on setting up this ASP.NET application to support logging in via external services.
                         </p>
-                    </div>
+                    </div>                   
```
Remove extra whitespace

hishamco commented on src/Modules/SimplCommerce.Module.Localization/Areas/Localization/Controllers/LocalizationApiController.cs:
```diff
@@ -47,23 +47,24 @@ public async Task<IActionResult> GetResources(string cultureId)
                 .ToListAsync();
 
-            if(cultureId != STANDARD_CULTURE_ID)
+            if (cultureId != STANDARD_CULTURE_ID)
```
Remove the extra space

afernandes replied:
Visual Studio is only respecting the spacing preferences in the **.editorconfig**
`csharp_space_after_keywords_in_control_flow_statements = true`

Should we alter this preference?

hishamco replied:
If is it like that just leave such comment and have a look to the others ..

hishamco commented on src/Modules/SimplCommerce.Module.Localization/Areas/Localization/Controllers/LocalizationApiController.cs:
```diff
@@ -47,23 +47,24 @@ public async Task<IActionResult> GetResources(string cultureId)
                 .ToListAsync();
 
-                foreach(var item in standardResources)
+                foreach (var item in standardResources)
```
and here

hishamco commented on src/Modules/SimplCommerce.Module.Localization/Areas/Localization/Controllers/LocalizationApiController.cs:
```diff
@@ -47,23 +47,24 @@ public async Task<IActionResult> GetResources(string cultureId)
+                foreach (var item in standardResources)
                 {
-                    if(resources.All(x => x.Key != item.Key))
+                    if (resources.All(x => x.Key != item.Key))
```
Please remove the extra space everywhere ..

hishamco commented on src/SimplCommerce.WebHost/Themes/CozaStore/Views/Shared/_Layout.cshtml:
```diff
@@ -254,9 +256,9 @@
     <script simpl-append-version="true" src="~/lib/toastr/build/toastr.min.js"></script>
     <script simpl-append-version="true" src="~/lib/bootstrap-star-rating/js/star-rating.js"></script>
-    @if (CultureInfo.CurrentCulture.Name == "vi-VN")
+    @if(CultureInfo.CurrentCulture.Name != "en-US")
```
Can we use the `GlobalConfiguration.DefaultCulture` instead?

afernandes replied:
It could be the GlobalConfiguration.DefaultCulture variable only if it is always en-US.

We should have another variable, example GlobalConfiguration.DefaultCultureUI to use in WorkContext.GetCurrentUser () for example.

It should be possible to select the default language for new users, or select to keep the language supported by the browser, but for this check it should be en-US, since for en-US no files are required.

hishamco replied:
I don't think so, default culture is `en-US` and default UI culture is `en-GB`. But If you are talking about user default one this can be set the property that localization middleware is using

hishamco commented on src/SimplCommerce.WebHost/Themes/CozaStore/Views/Shared/_Layout.cshtml:
```diff
@@ -1,7 +1,8 @@
 
 <!doctype html>
-<html>
+<html lang="@(Context.Features.Get<IRequestCultureFeature>().RequestCulture.UICulture.Name.ToLowerInvariant())">
```
Can we use `CultureInfo.CurrentUICulture.Name` instead?

hishamco commented on src/SimplCommerce.WebHost/Themes/CozaStore/Views/Shared/_Layout.cshtml:
```diff
@@ -254,9 +256,9 @@
+    @if(CultureInfo.CurrentCulture.Name != "en-US")
     {
-        <script simpl-append-version="true" src="~/lib/bootstrap-star-rating/js/locales/vi-vn.js"></script>
+    <script simpl-append-version="true" src="~/lib/bootstrap-star-rating/js/locales/@(CultureInfo.CurrentCulture.Name).js"></script>    
```
I was planned to do this in the past, please check if the star-rating using two letter culture for all the locales

afernandes replied:
I will do this in the next PR.
I'll create a standard function for handling multi-language files

funtionX("~/lib/bootstrap-star-rating/js/locales/{0}.js")
....
return "~/lib/bootstrap-star-rating/js/locales/pt-BR.js"
OR
return "~/lib/bootstrap-star-rating/js/locales/pt.js"

depending on which file exists

hishamco commented on src/SimplCommerce.WebHost/Themes/CozaStore/Views/Shared/_SelectLanguagePartial.cshtml:
```diff
@@ -0,0 +1,30 @@
+﻿@using Microsoft.AspNetCore.Localization
```
@thiennn IMHO we could move this to localization module to use it everywhere, instead of duplicated the code each time

thiennn replied:
noted. Thanks

hishamco commented on src/SimplCommerce.WebHost/Themes/SampleTheme/Views/Shared/_Layout.cshtml:
```diff
@@ -1,7 +1,8 @@
 
 <!doctype html>
-<html>
+<html lang="@(Context.Features.Get<IRequestCultureFeature>().RequestCulture.UICulture.Name.ToLowerInvariant())">
```
Use `CultureInfo.CurrentUICulture.Name` instead

hishamco commented on src/SimplCommerce.WebHost/wwwroot/themes/CozaStore/js/cozastore.js:
```diff
@@ -210,4 +210,21 @@
         $('.js-modal1').removeClass('show-modal1');
     });
 
+    $('a.lang-selector').on('click', function (e) {
```
Why we need this?

afernandes replied:
_SelectLanguagePartial of the "Coza" theme is different from the default and the function in main.js does not work for this theme.
![Coza localization](https://user-images.githubusercontent.com/867631/58650919-50ba8200-82e6-11e9-85dc-5a64e17ea60d.png)

jacek-gorgon commented:
The travis build seems to fail for precisely the same reason a fresh checkout fails for me - `global.json` breaks on newer SDKs. Remove `global.json`?

afernandes commented:
> The travis build seems to fail for precisely the same reason a fresh checkout fails for me - `global.json` breaks on newer SDKs. Remove `global.json`?

VS2017 supports at most version 2.2.108.

Global.json is important because each developer has a different version of the SDK installed on the machine. With global.json we force the use of the correct version.

afernandes commented:
@thiennn could you merge, or does anyone else have any suggestions?

thiennn commented:
Thanks @afernandes
